### DISCUSSION TITLE
Expose rule documentation link in the API

### DISCUSSION THREAD

**Discussion by silvenon** [biomejs/biome#4369] [Ideas]
Hi, my idea is to do whatever is necessary to make biomejs/biome-zed#58 possible. It would be nice for extensions to provide a way to link to rules so that we can read how to solve the problems that are underlined. It's faster than searching the docs.

**Comment by ematipico** [comment]
We already have that information. However I don't know how to create an action that opens a link.

If you're able to provide some code example in the wild, we might be able to implement it

**Reply by silvenon** [reply]
Is Zed's built-in ESLint integration of no use? Since it has the same code action.

**Reply by ematipico** [reply]
I suspect zed extension uses the same LSP server that the VScode extension uses. Or I might be wrong

**Reply by silvenon** [reply]
You might be right, I couldn't find a trace of such an implementation in Zed's codebase anywhere.

**Reply by silvenon** [reply]
So I guess you can close this if it's a duplicate of something else that was already reported.

**Reply by ematipico** [reply]
I think it's just a matter of checking how the eslint LSP does that.

When you have time and if you want to help, you could check its codebase and find how they implement it

**Reply by silvenon** [reply]
I may give it a shot because it sounds like an opportunity to learn something about LSPs. Thankfully it's just a nice-to-have so I don't feel pressure to succeed.

**Comment by ematipico** [comment]
I am going to close this. Our LSP already supports providing links when printing a diagnostic, check the screenshot from VSCode, the `biome(lint/style/useConst)` is actually a link to the documentation page.

<img width="771" alt="Screenshot 2024-11-04 at 13 54 44" src="https://github.com/user-attachments/assets/8bf3da8f-8a18-4fb4-93e7-da47753219be">

If zed doesn't show it, that's because it doesn't support it yet, so you could ask for a feature request.

**Reply by silvenon** [reply]
Yeah, that popup looks way better than what Zed shows, and if VS Code can do it, Zed should too, eventually.

If I find anything out I'll submit a PR wherever.
